### PR TITLE
Global Exception Fixes

### PR DESCRIPTION
**Pull request by shubhu0506**

### CODE REVIEW

File: src/main/java/com/searchservice/app/domain/service/InputDocumentService.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ private UploadDocumentUtil documentUploadResponse(String tableName, String paylo
 	public ThrottlerResponse addDocuments(boolean isNRT,String tableName, String payload) {
 
 		if (!manageTableServicePort.isTableExists(tableName))
-			throw new BadRequestOccurredException(400, tableName.split("_")[0] + " table doesn't exist");
+			throw new CustomExceptionHandler(400,HttpStatusCode.BAD_REQUEST_EXCEPTION,tableName.split("_")[0] + " table doesn't exist");
```
Comment (skdhitman) [inline src/main/java/com/searchservice/app/domain/service/InputDocumentService.java:67]: Please use the custom ErrorStatus from HttpStatusCode for table "does not exist"

File: src/main/java/com/searchservice/app/domain/utils/UploadDocumentUtil.java [hunk 2]
```diff
@@ -19,53 +19,72 @@
 @Data
 @Component
 public class UploadDocumentUtil {
-
-	private static final String APPLICATION_JSON = "application/json";
-
+	  private static String contentType="application/json";
```
Comment (skdhitman) [inline src/main/java/com/searchservice/app/domain/utils/UploadDocumentUtil.java:22]: can keep constant name in all capitals..

File: src/main/java/com/searchservice/app/domain/utils/UploadDocumentUtil.java [hunk 3]
```diff
@@ -19,53 +19,72 @@
...
 		String url = baseSearchUrl + "/" + tableName + "/update?";
+
 		url += "softCommit=true";
 
 		log.debug("SOFT COMMIT");
 
-		return processUploadDocumentRequest(client, body, url);
-	}
-	
-	private UploadDocumentSearchUtilRespnse processUploadDocumentRequest(OkHttpClient client, RequestBody body, String url) {
```
Comment (skdhitman) [inline src/main/java/com/searchservice/app/domain/utils/UploadDocumentUtil.java:55]: WHY this method is removed? Any reason?

File: src/main/java/com/searchservice/app/rest/errors/CustomExceptionHandler.java [hunk 4]
```diff
@@ -12,11 +14,14 @@
 @AllArgsConstructor
 @EqualsAndHashCode(callSuper = false)
 @Component
-public class TableNotUnderDeletionException extends RuntimeException{
-
+public class CustomExceptionHandler extends RuntimeException{
```
Comment (skdhitman) [inline src/main/java/com/searchservice/app/rest/errors/CustomExceptionHandler.java:17]: Would be better if we name it as "Exception" only.. like, CustomException/CustomSAASException

File: src/main/java/com/searchservice/app/rest/errors/RestControllerAdvice.java [hunk 5]
```diff
@@ -14,118 +14,25 @@
...
-
-		return new ResponseEntity<>(new RestApiErrorHandling(
-
-				HttpStatusCode.TABLE_NOT_UNDER_DELETION.getCode(), HttpStatusCode.TABLE_NOT_UNDER_DELETION,
-				exception.getExceptionMessage()), HttpStatus.BAD_REQUEST);
-	}
-	@ExceptionHandler(InvalidSKUOccurredException.class)
-	public ResponseEntity<Object> handleInvalidSKU(InvalidSKUOccurredException exception) {
+	@ExceptionHandler(CustomExceptionHandler.class)
+	public ResponseEntity<Object> handleGenricException(CustomExceptionHandler exception) {
```
Comment (skdhitman) [inline src/main/java/com/searchservice/app/rest/errors/RestControllerAdvice.java:27]: Generic spelling- optional change

**Review by skdhitman — CHANGES_REQUESTED** [review]

File: src/main/java/com/searchservice/app/domain/utils/UploadDocumentUtil.java (hunk 3 above)
Comment (shubhu0506) [reply src/main/java/com/searchservice/app/domain/utils/UploadDocumentUtil.java:55]: I didnot remove that.

Comment (skdhitman) [reply src/main/java/com/searchservice/app/domain/utils/UploadDocumentUtil.java:55]: It's there in main-dev branch, right? Pls check that properly

**Pull request merged**